src/setup_payload/AdditionalDataPayloadParser.cpp

```diff
@@ -0,0 +1,90 @@
+/**
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file describes a AdditionalData Payload parser based on the
+ *      CHIP specification.
+ */
+
+#include "AdditionalDataPayloadParser.h"
+#include "Base41.h"
+
+#include <iomanip>
+#include <math.h>
+#include <memory>
+#include <sstream>
+#include <string.h>
+#include <vector>
+
+#include <core/CHIPCore.h>
+#include <core/CHIPError.h>
+#include <core/CHIPTLVData.hpp>
+#include <core/CHIPTLVDebug.hpp>
+#include <core/CHIPTLVUtilities.hpp>
+#include <cstdlib>
+#include <iomanip>
+#include <iostream>
+#include <protocols/Protocols.h>
+#include <support/CodeUtils.h>
+#include <support/RandUtils.h>
+#include <support/SafeInt.h>
+#include <support/ScopedBuffer.h>
```

Comment:
Not used.  Nor is `SafeInt`.  In general, please just include the headers you actually need.

Reply:
fixed